src/app.js

```diff
@@ -1,5 +1,194 @@
+});
+
+app.get('/strings/first-character/:string', (req, res) => {
+  res.status(200);
+  res.json({ result: firstCharacter(req.params.string) });
+});
+
+app.get('/strings/first-characters/:string', (req, res) => {
+  res.status(200);
+  res.json({ result: firstCharacters(req.params.string, req.query.length) });
+});
+
+// Numbers
+
+app.get('/numbers/add/:a/and/:b', (req, res) => {
+  const a = parseInt(req.params.a);
+  const b = parseInt(req.params.b);
+  if (Number.isNaN(a) || Number.isNaN(b)) {
+    return res.status(400).json({ error: 'Parameters must be valid numbers.' });
+  }
+  return res.status(200).json({ result: add(a, b) });
+});
+
+app.get('/numbers/subtract/:a/from/:b', (req, res) => {
+  const a = parseInt(req.params.a);
+  const b = parseInt(req.params.b);
+  if (Number.isNaN(a) || Number.isNaN(b)) {
+    return res.status(400).json({ error: 'Parameters must be valid numbers.' });
+  }
+  return res.status(200).json({ result: subtract(b, a) });
+});
+
+app.post('/numbers/multiply', (req, res) => {
+  const { a } = req.body;
+  const { b } = req.body;
+
+  const n1 = parseInt(a, 10);
+  const n2 = parseInt(b, 10);
+  if (!a || !b) {
+    res.status(400).json({ error: 'Parameters "a" and "b" are required.' });
+  }
+  if (Number.isNaN(n1) || Number.isNaN(n2)) {
+    res.status(400).json({ error: 'Parameters "a" and "b" must be valid numbers.' });
+  }
+  res.status(200).json({ result: multiply(n1, n2) });
+});
+
+app.post('/numbers/divide', (req, res) => {
+  const { a } = req.body;
+  const { b } = req.body;
+
+  const n1 = parseInt(a, 10);
+  const n2 = parseInt(b, 10);
+
+  if (b === 0) {
+    res.status(400).json({ error: 'Unable to divide by 0.' });
+  }
+  if (a === 0) {
+    res.status(200).json({ result: divide(n1, n2) });
+  }
+  if (!a || !b) {
+    res.status(400).json({ error: 'Parameters "a" and "b" are required.' });
+  }
+  if (Number.isNaN(n1) || Number.isNaN(n2)) {
+    res.status(400).json({ error: 'Parameters "a" and "b" must be valid numbers.' });
+  }
+  res.status(200).json({ result: divide(n1, n2) });
+});
+
+app.post('/numbers/remainder', (req, res) => {
+  const { a } = req.body;
+  const { b } = req.body;
+
+  const n1 = parseInt(a, 10);
+  const n2 = parseInt(b, 10);
+
+  if (req.body.b === 0) {
+    res.status(400).json({ error: 'Unable to divide by 0.' });
+  }
+  if (req.body.a === 0) {
+    res.status(200).json({ result: remainder(req.body.a, req.body.b) });
+  }
+  if (!req.body.a || !req.body.b) {
+    res.status(400).json({ error: 'Parameters "a" and "b" are required.' });
+  }
+  if (Number.isNaN(n1) || Number.isNaN(n2)) {
+    res.status(400).json({ error: 'Parameters must be valid numbers.' });
+  }
+  res.status(200).json({ result: remainder(req.body.a, req.body.b) });
+});
+
+// Arrays
+
+app.post('/arrays/element-at-index/:a', (req, res) => {
+  const { array } = req.body;
+  const { a } = req.params;
+  res.status(200).json({ result: getNthElement(a, array) });
```

Comment:
No need to create the variables here since they are only used once, can be done clearly in a single line:
```
  res.status(200).json({ result: getNthElement(req.params.a, req.body.array) });
```